docs/source-pytorch/advanced/post-training quantization.rst

```diff
@@ -0,0 +1,158 @@
+
+The code changes that are required for Intel Neural Compressor are highlighted with comments in the line above.
+
+Pytorch Lightning model
+-----------------------
+::
+
+    import torch
+    from pytorch_lightning import LightningModule
+    from transformers import (
+        AutoConfig,
+        AutoModelForSequenceClassification,
+        AutoTokenizer,
+
+    # BERT Model definition
+    class GLUETransformer(LightningModule):
+        def __init__(self):
+            self.config = AutoConfig.from_pretrained(model_name_or_path, num_labels=num_labels)
+            self.model = AutoModelForSequenceClassification.from_pretrained(model_name_or_path, config=self.config)
+        def forward(self, **inputs):
+            return self.model(**inputs)
+
+    model = GLUETransformer(model_name_or_path=" Intel/bert-base-uncased-mrpc",)
+
+The fine-tuned model from Intel could be downloaded from `Intel Hugging Face repository <https://huggingface.co/Intel>`_.
+
+Accuracy-driven quantization config
+-----------------------------------
+Intel Neural Compressor supports accuracy-driven automatic tuning to generate the optimal INT8 model which meets a predefined accuracy goal. The default tolerance of accuracy loss in the accuracy criterion is 0.01. And the maximum trial number of quantization is 600. The user can specifically define their own criteria by:
+::
+    from neural_compressor.config import PostTrainingQuantConfig, TuningCriterion, AccuracyCriterion
+    accuracy_criterion = AccuracyCriterion(tolerable_loss=0.01)
+    tuning_criterion = TuningCriterion(max_trials=600)
+    conf = PostTrainingQuantConfig(approach="static", backend="default", tuning_criterion=tuning_criterion, accuracy_criterion=accuracy_criterion)
+
+The ''approach'' parameter in PostTrainingQuantConfig is defined by the user to make a choice from post-training static quantization and post-training dynamic by ''static'' or ''dynamic''.
+
+Conduct quantization
+--------------------
+The model is fed into Intel Neural Compressor and conduct model quantization:
+::
+    from neural_compressor.quantization import fit
+    q_model = fit(model=model.model, conf=conf, calib_dataloader=val_dataloader(), eval_func=eval_func)
+
+Users can define the evaluation function ''eval_func'' by themselves.
+
+At last, the quantized model can be saved by:
+::
+    q_model.save("./saved_model/")
+
+
+Hands-on Examples
+=================
+Based on the `given example code <https://pytorch-lightning.readthedocs.io/en/stable/notebooks/lightning_examples/text-transformers.html>`_, we show how Intel Neural Compressor conduct model quantization on PyTorch Lightning. We first define the basic config of the quantization process.
+::
+
+    from neural_compressor.quantization import fit as fit
+    from neural_compressor.config import PostTrainingQuantConfig, TuningCriterion, AccuracyCriterion
+
+    def eval_func_for_nc(model_n, trainer_n):
+        setattr(model, "model", model_n)

```

Comment:
this is a little ugly. @PenghuiCheng  is it what you mentioned before? that's the fx tracing saves temp model in other places, we have no way to do in-place model modification?